proteus/mprans/BoundaryConditions.py

```diff
@@ -188,6 +192,102 @@ def setAtmosphere(self, orientation=None, vof_air=1.):
         self.k_diffusive.setConstantBC(0.)
         self.dissipation_diffusive.setConstantBC(0.)
 
+    def setRigidBodyMoveMesh(self, body):
+        """
+        Sets boundary conditions for moving the mesh with a rigid body
+
+        Parameters
+        ----------
+        last_pos: array_like
+            last position of rigig body
+        h: array_like
+            displacement of the body
+        rot_matrix:
+            rotation matrix describing displament due to rotation between last
+            position and new position (3x3 array)
+
+        (!) if set manually, the input arrays should be updated externally
+            without loosing their memory address
+        """
+        def get_DBC_h(i):
+            def DBC_h(x, t):
+                x_0 = x-body.last_position
+                new_x_0 = np.dot(x_0, body.rotation_matrix)
+                hx = new_x_0-x_0+body.h
+                return hx[i]
+            return DBC_h
+        self.hx_dirichlet.uOfXT = get_DBC_h(0)
+        self.hy_dirichlet.uOfXT = get_DBC_h(1)
+        if body.nd > 2:
+            self.hz_dirichlet.uOfXT = get_DBC_h(2)
+
+
+    def setChMoveMesh(self, body):
+        self.hx_dirichlet.uOfXT = lambda x, t: body.hx(x, t)
+        self.hy_dirichlet.uOfXT = lambda x, t: body.hy(x, t)
+        self.hz_dirichlet.uOfXT = lambda x, t: body.hz(x, t)
+
+    def setTurbulentDirichlet(self, kVal, dissipationVal):
+        """
+        Sets only dirichlet conditions for turbulence at the boundary
```

Comment:
This is a rough approximation of a wall condition. You should highlight in the comments that the values provided here come from empirical approximations of the near wall turbulence, based on a notional free-stream velocity, defined by the general dimensions of the problem. It is not recommended to be used in the presence of more sophisticated wall boundary conditions